index.bs

```diff
@@ -257,28 +264,112 @@ spec:css-syntax-3;
   This document depends on the Infra Standard for a number of foundational concepts used in its
   algorithms and prose [[!INFRA]].
 
+
+  ### Infrastructure Algorithms ### {#sctn-infra-algorithms}
+
+  <!-- NOTE: '<h4>' maps to the same heading level as three hashs '###', thus using '<h5>' for the next section -->
+  <h5 id="algorithm-same-origin-with-ancestors" algorithm>Same-Origin with its Ancestors</h5>
+
   An [=environment settings object=] (|settings|) is <dfn noexport>same-origin with its
   ancestors</dfn> if the following algorithm returns `true`:
 
-  1.  If |settings| has no [=environment settings object/responsible document=],
-      return `false`.
+  <ol class="algorithm">
+    1.  If |settings| has no [=environment settings object/responsible document=],
+        return `false`.
+
+    2.  Let |document| be |settings|' [=environment settings object/responsible document=].
 
-  2.  Let |document| be |settings|' [=environment settings object/responsible document=].
+    3.  If |document| has no [=document/browsing context=], return `false`.
 
-  3.  If |document| has no [=document/browsing context=], return `false`.
+    4.  Let |origin| be |settings|' [=environment settings object/origin=].
 
-  4.  Let |origin| be |settings|' [=environment settings object/origin=].
+    5.  Let |current| be |document|'s [=document/browsing context=].
 
-  5.  Let |current| be |document|'s [=document/browsing context=].
+    6.  While |current| has a [=parent browsing context=]:
+
+        1.  Set |current| to |current|'s [=parent browsing context=].
+
+        2.  If |current|'s [=active document=]'s [=origin=] is not [=same origin=] with |origin|,
+            return `false`.
+
+    7.  Return `true`.
+  </ol>
+
+  ### Registry: Credential Type to Options Member Identifier Mappings ### {#sctn-registry-cred-type-to-options-member-identifier}
+
+  This registry maps [=credential types=] (i.e., {{Credential/[[type]]}} values) to the
+  [=dictionary member=] [=identifier=] used in
+  {{CredentialCreationOptions}} and {{CredentialRequestOptions}} (i.e., "options dictionaries")
+  by their specifications.
+
+  Note: This registry is used by the [=relevant credential interface objects=] algorithm.
+
+
+  <table class="complex data longlastcol">
+    <tr>
+        <th>Credential Type<br>
+        (in alphabetical order)</th>
+        <th>Options Member Identifier</th>
+        <th>Specification</th>
+        <th>Requestor Contact</th>
```

Comment:
What's the "Requestor Contact" for? The W3C or WICG in general won't be much help in modifying the specs, and the specs themselves have contact information in them. Would it make sense to just remove this column until and unless a credential type is registered by someone other than the editor of the type's specification?

Reply:
Other W3C registry specs have a "Requestor Contact" column, am just following that practice.  e.g.: 
* https://www.w3.org/TR/trace-context-protocols-registry/
* https://www.w3.org/TR/timing-entrytypes-registry/

I have not run across W3C guidelines for crafting registries -- is there such (@wseltzer) ?

Reply:
I don't know of W3C-wide guidelines. This'll probably get rejiggered into a formal [Registry](https://www.w3.org/2021/Process-20211102/#registries) before Credential Management goes to CR, but the [current example registry](https://w3c.github.io/tr-design/p2021mockup/dry-6.5.2.html) doesn't include any tables, so it's no help. I've filed https://github.com/w3c/tr-design/issues/283 for guidance.

Reply:
Ah, doh, I should've looked at the process document for guidance on a registry - thanks <grin type="sheepish">. That Registry section of the Process doc you cite would seem to qualify as W3C-wide registry guidelines.

So for now I suggest leaving the "Requestor Contact" column in there since it is in existing registries.

Reply:
I don't hate the "Requestor Contact" column; I just don't think the current contents are very useful. We can revisit it once there are a couple other examples of registries.